PR: Rename Related Components to Related Panes for clarity

CAM-Gerlach opened the pull request:
<!--- Before submitting your pull request, --->
<!--- please complete as much as possible of the following checklist: --->

# Pull Request

## Pull Request Checklist

* [x] Read and followed this repo's [Contributing Guidelines](https://github.com/spyder-ide/spyder-docs/blob/master/CONTRIBUTING.md)
* [x] Based your PR on the latest version of the correct branch (master or 4.x)
* [x] Checked your writing carefully for correct English spelling, grammar, etc
* [x] Described your changes and the motivation for them below


## Description of Changes

<!--- Describe what you've changed and why. --->

As originally noticed on #322 , the choice of name in the standard "Related components" heading is not ideal—"Related panes" is more clear and specific, shorter, and avoids confusion with any other use of components within the section titles, e.g. as on #322 . Therefore, we should change it. I've updated the style guide accordingly.


<!--- Thanks for your help making Spyder --->
<!--- and its documentation better for everyone! --->

ccordoba12 commented:
@CAM-Gerlach, this PR contains some images that I think don't belong here but are part of PR #322, right?

CAM-Gerlach commented:
Wow, thanks for the catch @ccordoba12 — some untracked files from my image optimization on that PR got picked up here and somehow I missed that at every step, I don't know how—that would have been a _really_ bad mistake. Fixed now, and I cleaned my workding dir so it won't happen again.

ccordoba12 reviewed (approved):
Thanks @CAM-Gerlach!
